addons/hr_expense/models/hr_expense.py

```diff
@@ -960,6 +960,8 @@ def message_new(self, msg_dict, custom_values=None):
             vals['account_id'] = account.id
 
         expense = super().message_new(msg_dict, dict(custom_values or {}, **vals))
+        # We need to know the total_amount to compute the quantity
+        expense.quantity = expense.total_amount / expense.price_unit if expense.product_has_cost and expense.price_unit else 1
```

Comment:
Why do we need to force the value of `quantity` here?
It's far from trivial, it deserves an explanation :nerd_face:

Reply:
You're right. I eventually kept it as simple as possible without changing external-to-the-problem issues